cli/azd/cmd/wire.go

```diff
@@ -26,90 +27,99 @@ func initConsole(
 func initDeployAction(
 	console input.Console,
 	ctx context.Context,
+	chain []middleware.Middleware,
 	o *internal.GlobalCommandOptions,
 	flags deployFlags,
 	args []string,
-) (actions.Action, error) {
+) (actions.MiddlewareEnabledAction, error) {
 	panic(wire.Build(DeployCmdSet))
 }
 
 func initInitAction(
 	console input.Console,
 	ctx context.Context,
+	chain []middleware.Middleware,
 	o *internal.GlobalCommandOptions,
 	flags initFlags,
 	args []string,
-) (actions.Action, error) {
+) (actions.MiddlewareEnabledAction, error) {
 	panic(wire.Build(InitCmdSet))
 }
 
```

Comment:
This also inverts the dependency graph where each action now takes a dependency on the middleware chain.  I know it is all just wrapped by wire code gen but it just feels odd.

Reply:
>inverts the dependency graph where each action now takes a dependency on the middleware chain

This is untrue, because `init` functions do not represent dependency graphs. They are dependency registrations.

You can see this clearly in how it manifests in `wire_gen.go`:

```golang
	cmdConfigGetAction := newConfigGetAction(userConfigManager, formatter, writer, args)
	middlewareRunner := middleware.NewMiddlewareRunner(chain, cmdConfigGetAction)
	return middlewareRunner, nil
```